src/nvme/mi-mctp.c

```diff
@@ -220,6 +226,200 @@ static bool nvme_mi_mctp_resp_is_mpr(void *buf, size_t len,
 	return true;
 }
 
+static int nvme_mi_mctp_aem_fd(struct nvme_mi_ep *ep)
+{
+	struct nvme_mi_transport_mctp *mctp;
+
+	if (ep->transport != &nvme_mi_transport_mctp) {
+		errno = EINVAL;
+		return -1;
+	}
```

Comment:
now we know it's a mctp transport, just assign to `mctp` here, and you don't need to refer to `->transport_data` multiple times below.